dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2246,50 +2246,79 @@ bool RtpsUdpDataLink::separate_message(EntityId_t entity)
 namespace {
 
 struct BundleHelper {
+  const size_t initial_size =
```

Comment:
```suggestion
  static const size_t initial_size =
```

Reply:
Correct me if I'm wrong, but it seems to me that a `const` and a `static const` class member would have the same result.

Reply:
It's static because it's a property of the class and not a specific object.  Only static data members that are integral/enum types can be initialized in the class definition.

Reply:
That is to say that `const` would imply the information is common to instances of the class, which is same thing as `static`. I'm not opposed to always saying `static const` and I'll change it, but I'm just curious. I guess it could be stored with each object, but that seems like a massive optimization for a compiler to miss.

Reply:
Actually I guess if it's a const member that initalized by the constructor, then I guess it's not common to all instances. Then I think it's a optimization the compiler should still make if that's not the case but I guess it's not as simple as I was thinking.

Reply:
As long as what you're proposing compiles on all supported platforms, I'd be OK with it.  But I don't think it will.

The use of `static` here is different than just `const`.  A `const` member can be assigned in the constructor (can depend on constructor arguments).  A `static const` (which is integral/enum and in-class initialized) is completely determined at compile time.

Reply:
I never reloaded the page, so I didn't see you're first response, so I changed my mind just thinking though about it, but the point about the initialization in definition is enough. I added `static` to the code locally and will try to keep this in mind going forward.